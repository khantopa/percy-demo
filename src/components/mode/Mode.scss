@import '../../styles/main.scss';

.mode {
  width: 300px;
  height: 300px;
  box-sizing: border-box;
  @include absPosition(50%, 25%);
  transform: translateX(-50%);
  text-align: center;
  padding: 10px;
  &_head {
    margin-bottom: 20px;
    color: white;
  }
  &_group {
    margin-top: 80px; 
    color: white;
    &_button {
      width: 100%;
      margin: 40px auto;
      @extend .button;
    }
  }
}


@media screen and (max-width: $mobile-width) {
  .mode {
    padding: 20px;
    width: 100vw;
    height: 300px;
  }
}