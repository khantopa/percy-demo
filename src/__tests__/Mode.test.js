import { Link } from 'react-router-dom';
import Mode from '../components/mode/Mode';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Mode/> );
});

describe('<Mode>', () => {
  
  it('should <Board> defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders <Mode> without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  
  it('renders two <Link>', () => {
    expect(wrapper.find(Link)).toHaveLength(2);
  });
  
});
