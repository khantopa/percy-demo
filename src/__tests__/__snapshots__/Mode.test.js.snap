// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mode> renders <Mode> without crashing 1`] = `
<div
  className="mode"
>
  <h2
    className="mode_head"
  >
    Select your play mode
  </h2>
  <div
    className="mode_group"
  >
    <Link
      className="mode_group_button"
      key="Player vs Computer"
      role="button"
      to={
        Object {
          "pathname": "/play",
          "state": Object {
            "mode": "Player vs Computer",
          },
        }
      }
    >
      Player vs Computer
    </Link>
    <Link
      className="mode_group_button"
      key="Computer vs Computer"
      role="button"
      to={
        Object {
          "pathname": "/play",
          "state": Object {
            "mode": "Computer vs Computer",
          },
        }
      }
    >
      Computer vs Computer
    </Link>
  </div>
</div>
`;
